library/alloc/src/vec/drain.rs

```diff
@@ -131,15 +126,45 @@ impl<T, A: Allocator> Drop for Drain<'_, T, A> {
             }
         }
 
-        // exhaust self first
-        while let Some(item) = self.next() {
-            let guard = DropGuard(self);
-            drop(item);
-            mem::forget(guard);
+        let iter = mem::replace(&mut self.iter, (&mut []).iter());
+        let drop_len = iter.len();
+        let drop_ptr = iter.as_slice().as_ptr();
```

Comment:
Specifically, Miri says that in this line `as_slice` points to dangling memory (in some cases, here triggered by the `test_replace_range` testcase)

Reply:
I'm under the impression that the iter should be valid at all times, but I'll have a closer look at the test-case.

Reply:
(Continuing in https://github.com/rust-lang/rust/issues/91772)